analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -222,4 +222,677 @@ public class Test {
+    public void IdentifierToken_Type_MethodDeclaration(string returnType, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+
+            public class Test
+            {
+                public {{returnType}} M() => default;
+            }
+            """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("[t:Int32]", false)]
+    [DataRow("[u:System].Int32", true)]
+    public void IdentifierToken_Type_OperatorDeclaration(string returnType, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+
+            public class Test
+            {
+                public static {{returnType}} operator + (Test x) => default; 
+            }
+            """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("public static explicit operator [t:Int32](Test x)", false)]
+    [DataRow("public static explicit operator [u:System].Int32(Test x)", true)]
+    [DataRow("public static implicit operator [t:Int32](Test x)", false)]
+    [DataRow("public static implicit operator [u:System].Int32(Test x)", true)]
+    public void IdentifierToken_Type_ConversionOperatorDeclaration(string conversion, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+
+            public class Test
+            {
+                {{conversion}} => default; 
+            }
+            """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("public [t:Int32] Prop { get; set; }", false)]
+    [DataRow("public [t:Int32] Prop { get => 1; set { } }", false)]
+    [DataRow("public [t:Int32] this[int index] { get => 1; set { } }", false)]
+    [DataRow("public event [t:EventHandler] E;", false)]
+    [DataRow("public event [t:EventHandler] E { add { }  remove { } }", false)]
+    public void IdentifierToken_Type_PropertyDeclaration(string property, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+
+            public class Test
+            {
+                {{property}}
+            }
+            """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("[t:Int32]", false)]
+    [DataRow("[u:System].Int32", true)]
+    public void IdentifierToken_Type_LocalFunction(string returnType, bool allowSemanticModel = true) =>
+        ClassifierTestHarness.AssertTokenTypes($$"""
+            using System;
+
+            public class Test
+            {
+                public void M()
+                {
+                    {{returnType}} LocalFunction() => default;
+                }
+            }
+            """ /*, allowSemanticModel */);
+
+    [DataTestMethod]
+    [DataRow("[t:Int32]", false)]
+    [DataRow("[u:System].Int32", true)]
+    public void IdentifierToken_Type_ParenthesizedLambda(string returnType, bool allowSemanticModel = true) =>
```

Comment:
Missing:
* Attribute `[[t:Obsolete]]`
* ExplicitInterfaceSpecifier `[t:IEnumerable].GetEnumerator() => this;`
* UsingDirectiveSyntax `using [u:System];`
* BaseNamespaceDeclarationSyntax `namespace [u:A];` and `namspace [u:A] { }`

Reply:
Will you add them in a following PR?

Reply:
I added the first two as they were missing in the tests and implementation. The remaining two are under test already (IdentifierToken_Usings and IdentifierToken_Namespaces) but missing in the "IsTypeContext" implementation.